### PR TITLE
fix: Don't panic when recovering while combining Pratt matches

### LINKED ISSUE #1035: unreachable code: Error recovery happens outside precedence parsing
https://nomic-labs.sentry.io/issues/5216379985/

> Document Symbol Error: Error: internal error: entered unreachable code: Error recovery happens outside precedence parsing

### PR DESCRIPTION
**Pull request by Xanewok**
Closes #1035

Fow now this simply stops panicking in the recover path but we do not attempt any reduction post recovery.

### CODE REVIEW

File: crates/solidity/testing/snapshots/cst_output/SourceUnit/pratt_precedence_recovery/generated/0.4.11-failure.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                      - (statements꞉ Statements): # "\t\t2 +" (29..34)
+                          - (item꞉ Statement) ► (variant꞉ ExpressionStatement) ► (expression꞉ Expression): # "\t\t2 +" (29..34)
+                              - (variant꞉ DecimalNumberExpression): # "\t\t2" (29..32)
+                                  - (leading_trivia꞉ Whitespace): "\t\t" # (29..31)
+                                  - (literal꞉ DecimalLiteral): "2" # (31..32)
+                              - (AdditiveExpression): # " +" (32..34)
+                                  - (leading_trivia꞉ Whitespace): " " # (32..33)
+                                  - (operator꞉ Plus): "+" # (33..34)
+                      - (MISSING): "" # (34..34)
+                      - (close_brace꞉ CloseBrace): " " # (34..35)
```
Comment (OmarTawfik) [inline crates/solidity/testing/snapshots/cst_output/SourceUnit/pratt_precedence_recovery/generated/0.4.11-failure.yml:54]: Looks like this uncovered another bug with delimiter recovery. I'll double check.

Comment (OmarTawfik) [reply crates/solidity/testing/snapshots/cst_output/SourceUnit/pratt_precedence_recovery/generated/0.4.11-failure.yml:54]: I think the PRATT parser in this branch is producing an incomplete tree.
I created #1040 which will make the bug fail this specific test.

Comment (Xanewok) [reply crates/solidity/testing/snapshots/cst_output/SourceUnit/pratt_precedence_recovery/generated/0.4.11-failure.yml]: Small mistake on my part, I forgot to also add `.chain(skipped.nodes)`. Thanks for noticing that!

**Comment by Xanewok** [comment]
Rebased and should be ready for re-review.

**Review by OmarTawfik — APPROVED** [review]

**Comment by OmarTawfik** [comment]
CI seems to be unblocked now.

**Pull request merged**